portal-backend/depmap/compound/utils.py

```diff
@@ -0,0 +1,59 @@
+from depmap import data_access
+from depmap.compound import new_dose_curves_utils
+from depmap.compound.models import (
+    DRCCompoundDataset,
+    DRCCompoundDatasetWithNamesAndPriority,
+)
+
+
+def get_compound_dataset_with_name_and_priority(drc_dataset: DRCCompoundDataset):
+    auc_dataset_display_name = data_access.get_dataset_label(
+        drc_dataset.auc_dataset_given_id
+    )
+    priority = data_access.get_dataset_priority(drc_dataset.auc_dataset_given_id)
+    viability_dataset_display_name = data_access.get_dataset_label(
+        drc_dataset.viability_dataset_given_id
+    )
+
+    with_names_and_priority = DRCCompoundDatasetWithNamesAndPriority(
+        drc_dataset_label=drc_dataset.drc_dataset_label,
+        viability_dataset_given_id=drc_dataset.viability_dataset_given_id,
+        replicate_dataset=drc_dataset.replicate_dataset,
+        auc_dataset_given_id=drc_dataset.auc_dataset_given_id,
+        display_name=drc_dataset.display_name,
+        priority=priority,
+        auc_dataset_display_name=auc_dataset_display_name,
+        viability_dataset_display_name=viability_dataset_display_name,
+    )
+
+    return with_names_and_priority
+
+
+def dataset_exists_with_compound_in_auc_and_rep_datasets(
+    drc_dataset: DRCCompoundDataset, compound_label: str, compound_id: str
+) -> bool:
+    """ 
+    We only want to load the dose curves and heatmap if the auc and replicate datasets all exist with the compound.
+    The Heatmap relies on dose curve data to load the curve params (hidden columns by default) into the table.
```

Comment:
I'd really like to believe that we always have Viability data when we have AUC data, but that _might_ not be safe to assume so I am glad you're explicitly checking 😅